### PR TITLE
Feature/cleanup

### PR DESCRIPTION
**Pull request by kefniark**
# Description
- Add namespaces (change the folder structure)
- Linting and remove bunch of warnings (Visual Studio & Resharper are our friends)
- Add lot of null check

Few open issues are probably concerned by this PR https://github.com/mage/mage-sdk-unity/issues/5 https://github.com/mage/mage-sdk-unity/issues/7 https://github.com/mage/mage-sdk-unity/issues/14 https://github.com/mage/mage-sdk-unity/issues/18 https://github.com/mage/mage-sdk-unity/issues/19

**P.S.** This PR move and rename lot of method, updating from a previous version is not smooth and need some modules modifications 
# TODO Later
- Remove usage of reflection for the module instantiation
- Remove the async class, we can do without (current usage are limitated and need to be rewritten anyway)
- Change the event emitter to use proper c# event

### CODE REVIEW

**Comment by MiLk** [comment]
It would be a good idea to start doing release and creating tags on this project.
It would avoid breaking projects when the API changes.

**Comment by nullorvoid** [comment]
At the moment it looks like this is a per project branch structure, but at some point we should start tagging and releasing them I agree.  @ronkorving @AlmirKadric do we have a path to go down to start doing this?

**Comment by ronkorving** [comment]
What do you mean by "a path"? Just tag em and bag em :) Anyone willing to take responsible ownership can be a release manager.

**Comment by christopherobin** [comment]
You'll want to port the latest changes made by @AlmirKadric in his PR #35 especially the bits about headers and cookies

**Comment by kefniark** [comment]
Yes, we are planning with @nullorvoid to rewrite this network part and make it testable. So I will backport the last PRs at that time (ipv6, header, cookies,...)

File: EditorScript/UnityEditorPlayMode.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
+﻿#if UNITY_EDITOR
+
+using UnityEditor;
+
+namespace Wizcorp.MageSDK.Editor
+{
+	[InitializeOnLoad]
+	public static class UnityEditorPlayMode
```
Comment (nullorvoid) [inline EditorScript/UnityEditorPlayMode.cs:8]: Would be nice for some docs on this

**Comment by nullorvoid** [comment]
I'm ok with merging this PR to get things moving (it's not against develop) @AlmirKadric @ronkorving

**Pull request merged**